brave/brave-ios
# Fix #1386: Add rewards reporting to support media publishers and activity

- Panel now shows proper channels, names, etc. for supported publishers (YouTube, Vimeo, Twitch, GitHub, Twitter)
- Panel now shows proper activity for video related publishers (YouTube, Vimeo, Twitch)
- Now "hides" the tab (therefore computing activity) when the rewards panel is shown and "shows" it when its hidden
- Does the same when full-screen modals appear (tab is technically "hidden")
- Shows correct verified badge for publishers
- Supports notification count, however UI isn't added yet to show the red dot yet



## Submitter Checklist:



## Test Plan:

Prefix: Must remove `NO_REWARDS` from config for rewards to appear in the toolbar

1. Visit a normal publisher website, wait 8+ seconds
1. Open rewards panel: should show activity %
1. Visit a verified publisher website, should see badge in the toolbar
1. Open rewards panel: should show verified
1. Visit each video media publisher: YouTube, Vimeo, Twitch, load a video, and watch for 8+ seconds
1. Open rewards panel: should show `<user> on <platform>` as the name, activity should be non-zero
1. Visit a user profile of the other media publishers: GitHub, Twitter, Reddit and wait 8+ seconds on the page
1. Open rewards panel: should show `<user> on <platform>` as the name, **activity should be zero** as these publishers do not accumulate duration activity

### Screenshots:

| Twitter | Twitch |
| -- | -- |
| ![Simulator Screen Shot - iPhone 8 - 2019-08-13 at 12 25 11](https://user-images.githubusercontent.com/529104/62959148-1a10e580-bdc6-11e9-8f41-38ead36e42f3.png) | ![Simulator Screen Shot - iPhone 8 - 2019-08-13 at 12 29 29](https://user-images.githubusercontent.com/529104/62959164-2432e400-bdc6-11e9-86b8-d2e2b10b69f1.png) |
| YouTube | GitHub | 
| ![Simulator Screen Shot - iPhone 8 - 2019-08-13 at 12 28 21](https://user-images.githubusercontent.com/529104/62959176-2a28c500-bdc6-11e9-8e61-bd9c79bbf431.png) | ![Simulator Screen Shot - iPhone 8 - 2019-08-13 at 12 29 14](https://user-images.githubusercontent.com/529104/62959189-33199680-bdc6-11e9-99ef-2c317e949b3f.png) 
| Auto-Contribute | |
| ![Simulator Screen Shot - iPhone 8 - 2019-08-13 at 12 28 54](https://user-images.githubusercontent.com/529104/62959243-52182880-bdc6-11e9-9997-cdb2623d67ec.png) |

## Reviewer Checklist:
